mtags/src/main/scala-3/scala/meta/internal/pc/HoverProvider.scala

```diff
@@ -49,8 +51,43 @@ object HoverProvider:
     val exprTp = typeFromPath(enclosing)
     val exprTpw = exprTp.widenTermRefExpr
 
-    if tp.isError || tpw == NoType || tpw.isError || path.isEmpty then
-      ju.Optional.empty()
+    /**
+     * Check if the given `sourcePos` is on the name of enclosing tree.
+     * ```
+     * // For example, if the postion is on `foo`, returns true
+     * def foo(x: Int) = { ... }
+     *      ^
+     *
+     * // On the other hand, it points to non-name position, return false.
+     * def foo(x: Int) = { ... }
+     *  ^
+     * ```
+     * @param path - path to the position given by `Interactive.pathTo`
+     */
+    def isHoveringOnName(path: List[Tree], sourcePos: SourcePosition): Boolean =
+      def contains(tree: Tree): Boolean = tree match
+        case tree: NameTree =>
+          source.fold(false) { s =>
+            SourceTree(tree, s).namePos.contains(sourcePos)
+          }
+        // TODO: check the positions for NamedArg and Import
```

Comment:
@tanishiking should it have been addressed in this PR before merging?

Reply:
The drawback of not checking these trees is: we show a hover tooltip if the cursor is on

- ` = ` part of named argument
- ` as ` part of import statement

which aren't big problems, so I believe we can address these in another PR (maybe we should open another issue)

<img width="431" alt="Screen Shot 2022-04-08 at 12 39 08" src="https://user-images.githubusercontent.com/9353584/162358120-3fadf1cf-0e1d-46b0-be50-1194ca5025f3.png">

<img width="332" alt="Screen Shot 2022-04-08 at 12 39 00" src="https://user-images.githubusercontent.com/9353584/162358119-8bca8fcd-fff9-468f-a005-c68c572dcdec.png">

Reply:
For sure this is not a big issue :+1: